src/Npgsql/NpgsqlConnection.cs

```diff
@@ -919,12 +919,14 @@ public TextWriter BeginTextImport(string copyFromCommand)
         /// in <paramref name="copyToCommand"/>.
         /// </summary>
         /// <param name="copyToCommand">A COPY TO STDOUT SQL command</param>
+        /// <param name="parameters">TODO</param>
         /// <returns>
         /// A TextReader that can be used to read textual data.</returns>
         /// <remarks>
         /// See http://www.postgresql.org/docs/current/static/sql-copy.html.
         /// </remarks>
-        public TextReader BeginTextExport(string copyToCommand)
+        // TODO: Better API for this...
+        public TextReader BeginTextExport(string copyToCommand, NpgsqlParameterCollection parameters = null)
```

Comment:
To keep the API consistent, it may be better to have this method (and the others) on NpgsqlCommand rather than on NpgsqlConnection. This way you instantiate an NpgsqlCommand, assign all the parameters just like with a normal command, but instead of calling `ExecuteReader()` you call `ExecuteTextExport()`. This would also allow easier code sharing with other NpgsqlCommand codepaths.

We can still keep a parameterless version on NpgsqlConnection to make things simpler for the common case, and also for backwards compat. It may also make sense to rename from `BeginTextExport()` to `ExecuteTextExport()` for consistency.